### ISSUE TITLE
Use TVM to generate code for some operators or chains of operators

### ISSUE THREAD

**Issue by gussmith23** [gussmith23/glenside#58]
cc @hypercubestart as a future project of potential interest, given the TVM overlap.

In our networks, we generally care about "tensorizing" (aka converting to hardware) the convolutions and fully connected layers, as they're the most computationally intense and acceleratable. There are many other operators in a network, however; batch norms, activation functions, pooling, softmax, etc. While we will slowly start supporting some of these operators in hardware (i.e. activation functions after convolutions), some of the operators are more likely to be computed on the ASIC's RISCV CPU for the foreseeable future (like softmax). For these operators, it would be nice to just leverage TVM to generate software kernels for us to run on the CPU. This would hopefully give us okay performance on these operators, even though they're not implemented in hardware.

This is what I think is involved in this task:
- [ ] Investigate whether we can use TVM to generate RISCV kernels which will run on the chip. Specifically, this means we'd need to generate kernels that don't rely on any TVM runtime, as we will not have the TVM runtime on-chip (at least not at the moment).
   - Side-note: we can look into what it would take to run the TVM runtime on chip.
 - [ ] Ensure we can go from Glenside code back to Relay/TVM.
   - This could be as simple as, when ingesting models from Relay, simply save a pointer to the Relay object as part of the eclass analysis for the generated node. 
   - If a model is NOT ingested from Relay, then we'd need to support generating Relay/TVM from Glenside constructs.
 - [ ] During codegen, invoke TVM codegen for these operators and include the generated kernels in the output.
   - Perhaps we can use the C backend? We can generate kernels in C and then compile them to RISCV?
